network/src/peers/peers.rs

```diff
@@ -142,42 +142,33 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
     }
 
     ///
-    /// Broadcasts a connection request to all default bootnodes of the network.
+    /// Broadcasts a connection request to all the supplied addresses.
     ///
-    /// This function attempts to reconnect this node server with any bootnode peer
-    /// that this node may have failed to connect to.
-    ///
-    /// This function filters out any bootnode peers the node server is
+    /// This function filters out any peers the node server is
     /// either connnecting to or already connected to.
     ///
-    async fn connect_to_bootnodes(&self) {
+    pub async fn connect_to_addresses(&self, addrs: &[SocketAddr]) {
         // Local address must be known by now.
         let own_address = self.local_address().unwrap();
 
-        // Iterate through each bootnode address and attempt a connection request.
-        for bootnode_address in self
-            .config
-            .bootnodes()
-            .iter()
-            .filter(|peer| **peer != own_address)
-            .copied()
-        {
+        for node_addr in addrs.iter().filter(|addr| **addr != own_address).copied() {
             let node = self.clone();
-            if node.peer_book.is_connected(bootnode_address) {
-                return;
+
+            if node.peer_book.is_connected(node_addr) {
+                break;
```

Comment:
nothing wrong with this PR here, but I see room for a small improvement: this check could be moved above the `node` binding, so that we avoid its clone; sure, it's just an `Arc`, but still, it's not free and avoidable